userguide/object_detection/data-preparation.md

```diff
@@ -0,0 +1,134 @@
+# IG02 Data Preparation
+
+In this section, we will show you how to download a publicly available dataset
+and load it into an SFrame. This will allow you to try out the Object Detection
+toolkit for yourself. To get this dataset into the format expected by our
+toolkit, we will rely on many useful SFrame functions.
+
+*Note: Please make sure that you have Turi Create 4.3 or above for these steps*
+
+The dataset that we will use is [INRIA Annotations for Graz-02
+(IG02)](https://lear.inrialpes.fr/people/marszalek/data/ig02/). Our goal is to
+make a detector that can spot bikes and cars. Go ahead and click this link and
+download `ig02-v1.0-bikes.zip` and `ig02-v1.0-cars.zip`. In this tutorial, we
+will assume that you have unzipped these files into `~/Downloads/ig02`. That
+is, you should have the following data structure:
+
+```
+~/Downloads/ig02/
+    bike/
+        bike_001.image.png
+        bike_001.mask.0.png
+        bike_001.mask.1.png
+        bike_001.mask.2.png
+        bike_001.mask.3.png
+        bike_002.image.png
+        ...
+    cars/
+        carsgraz_001.image.png
+        carsgraz_001.mask.0.png
+        carsgraz_002.image.png
+        ...
+```
+
+This dataset contains masks marking the exact pixels belonging to each object
+instance. Each image may have several masks to indicate the presence of
+multiple objects. Red pixels indicate object presence and green indicate
+presence but with [occlusion](https://en.wikipedia.org/wiki/Occultation). For
+object detection, we will convert this richly annotated data to bounding boxes.
+Here is a snippet to turn this into an SFrame:
+
+```python
+import turicreate as tc
+import numpy as np
+import glob
+import os
+
+# Change if applicable
+ig02_path = '~/Desktop/ig02'
+
+# Load all images in random order
+raw_sf = tc.image_analysis.load_images(ig02_path, recursive=True,
+                                       random_order=True)
+
+# Split file names so that we can determine what kind of image each row is
+# E.g. bike_005.mask.0.png -> ['bike_005', 'mask']
+info = raw_sf['path'].apply(lambda path: os.path.basename(path).split('.')[:2])
+
+# Rename as 'name', 'type', 'mask_index'
```

Comment:
I don't think you want to mention 'mask_index' here. Also I would probably say "Rename the columns to" rather than "Rename as".

Reply:
Very good catch! `mask_index` is old stuff, indeed. I agree with your other suggestion too.